tests/interfaces/test_autograd.py

```diff
@@ -811,9 +796,165 @@ def cost(a, b, c, weights):
         assert np.allclose(np.hstack(res), expected, atol=tol, rtol=0)
 
         # Check that the gradient was computed
-        # for all parameters in circuit2 (i.e., wrt=None)
-        assert spy.call_args_list[0][1]["wrt"] == None
+        # for all parameters in circuit2
+        assert circuit2.get_trainable_args() == {0, 1}
 
         # check that the parameter-shift rule was not applied
         # to the first parameter of circuit1
-        assert spy.call_args_list[1][1]["wrt"] == {1, 2}
+        assert circuit1.get_trainable_args() == {1, 2}
+
+    def test_non_diff_not_a_variable(self, capsys):
+        """Test that an argument marked as non-differentiable
+        is not wrapped as a variable."""
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev, interface="autograd")
+        def circuit(x, y, z):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=0)
+            qml.RZ(z, wires=0)
+            print(type(x), type(y), type(z))
+            return qml.expval(qml.PauliZ(0))
+
+        x = 1
+        y = qml.numpy.array(2, requires_grad=False)
+        z = 3
+
+        res = circuit(x, y, z)
+
+        assert circuit.get_trainable_args() == {0, 2}
+
+        assert isinstance(circuit.arg_vars[0], qml.variable.Variable)
+        assert not isinstance(circuit.arg_vars[1], qml.variable.Variable)
+        assert isinstance(circuit.arg_vars[2], qml.variable.Variable)
+
+        assert circuit.arg_vars[0] != x
+        assert circuit.arg_vars[1] == y
+        assert circuit.arg_vars[2] != z
+
+        captured = capsys.readouterr()
+        assert (
+            "<class 'pennylane.variable.Variable'> <class 'pennylane.numpy.tensor.tensor'> <class 'pennylane.variable.Variable'>"
+            in captured.out
+        )
+
+    @pytest.mark.parametrize("w", [[0, 1], [1, 0]])
+    def test_non_diff_wires_argument(self, w, tol):
+        """Test that passing wires as a non-differentiable positional
+        argument works correctly."""
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface="autograd")
+        def circuit(wires, params):
+            qml.Hadamard(wires=wires[0])
+            qml.CNOT(wires=[wires[0], wires[1]])
+            qml.RX(params[0], wires=wires[0])
+            qml.RY(params[1], wires=wires[1])
+            qml.CNOT(wires=[wires[1], wires[0]])
+            return qml.expval(qml.PauliZ(0))
+
+        params = qml.numpy.array([0.6, 0.2])
+        wires = qml.numpy.array(w, requires_grad=False)
+
+        a, b = params
+        if w == [0, 1]:
+            expected_res = np.cos(a) * np.cos(b)
+            expected_grad = [-np.cos(b) * np.sin(a), -np.cos(a) * np.sin(b)]
+        elif w == [1, 0]:
+            expected_res = 0
+            expected_grad = [0, 0]
```

Comment:
Essentially, it is checking the QNode result and gradient is correct, depending on the wire permutations. I've made two updates to this test to make it clearer:

* I've modified the circuit so that both wire permutations have non-zero expectation values.

* I've removed this if statement, and have included the expected results in the decorator:

  ```python
  a = 0.6
  b = 0.2
  test_data = [
      ([0, 1], np.cos(2*a) * np.cos(b), [-2 * np.cos(b) * np.sin(2*a), -np.cos(2*a) * np.sin(b)]),
      ([1, 0], -np.cos(b) * np.sin(b), [0, -np.cos(b) ** 2 + np.sin(b) ** 2]),
  ]
  
  @pytest.mark.parametrize("w, expected_res, expected_grad", test_data)
  def test_non_diff_wires_argument(self, w, expected_res, expected_grad, tol):
      """Test that passing wires as a non-differentiable positional
      argument works correctly, and results in the expected QNode result and gradient."""
      dev = qml.device("default.qubit", wires=2)
  
      @qml.qnode(dev, interface="tf")
      def circuit(wires, params):
          qml.Hadamard(wires=wires[0])
          qml.CNOT(wires=[wires[0], wires[1]])
          qml.RX(params[0], wires=wires[0])
          qml.RY(params[1], wires=wires[1])
          qml.CNOT(wires=[wires[1], wires[0]])
          qml.RX(params[0], wires=wires[0])
          qml.RY(params[1], wires=wires[1])
          return qml.expval(qml.PauliZ(0))
  
      params = tf.Variable([0.6, 0.2])
  
      with tf.GradientTape() as tape:
          res = circuit(w, params)
  
      assert circuit.get_trainable_args() == {1}
      assert np.allclose(res, expected_res, atol=tol, rtol=0)
  
      grad = tape.gradient(res, params)
  
      assert circuit.get_trainable_args() == {1}
      assert np.allclose(grad, expected_grad, atol=tol, rtol=0)
  ```